__tests__/articles.test.js

```diff
@@ -45,4 +45,81 @@ describe("GET/api/articles/:article_id", () => {
 
         expect(returnedArticle).toEqual(expectedArticle)
     })
+})
+
+
+describe("GET/api/articles", () => {
+    test("200 - responds with a 200 status and an array of all 13 topics", async () => {
+        const result = await request(app).get("/api/articles").expect(200)
+        const articlesArray = result.body.articles
+
+        expect(Array.isArray(articlesArray)).toBe(true)
+        expect (articlesArray.length).toBe(13)
+    })
+
+    test("200 - all articles have the appropriate keys / values", async () => {
+        const result = await request(app).get("/api/articles").expect(200)
+        const articlesArray = result.body.articles
+
+        articlesArray.forEach( (article) => {
+            expect(article).toMatchObject(
+                {
+                    author: expect.any(String),
+                    title: expect.any(String), 
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String),
+                    comment_count: expect.any(Number),
+                }
+            )
+        });
+    })
+
+    test("200 - all articles have the correct comment count", async () => {
+        const result = await request(app).get("/api/articles").expect(200)
+        const articlesArray = result.body.articles
+
+        articlesArray.forEach( (article) => {
+            const id = article.article_id
+            const commentCount = article.comment_count
+
+            switch (id) {
+                case 1: 
+                    expect(commentCount).toBe(11)
+                    break;
+                case 3: 
+                    expect(commentCount).toBe(2)
+                    break;
+                case 5: 
+                    expect(commentCount).toBe(2)
+                    break;
+                case 6: 
+                        expect(commentCount).toBe(1)
+                        break;
+                case 9: 
+                        expect(commentCount).toBe(2)
+                           break;
+                default:
+                    expect(commentCount).toBe(0)
+            }
+        });
+    })
+
+    test("200 - the articles are sorted by the created_at value, in descending order", async () => {
+        const result = await request(app).get("/api/articles").expect(200)
+        const articlesArray = result.body.articles
+
+        expect(articlesArray).toBeSortedBy("created_at", { descending: true })
+    })
+
+    test("200 - the articles do not have a body property", async () => {
+        const result = await request(app).get("/api/articles").expect(200)
+        const articlesArray = result.body.articles
+
+        articlesArray.forEach( (article) => {
+            expect(typeof article.body).toBe('undefined')
```

Comment:
Could this be done in your other forEach?

Reply:
Yeah, maybe. I just thiought it was 2 separate issues - 1: THe object is the right shape, 2: The values are right.